symengine/univariate_int.h

```diff
@@ -0,0 +1,82 @@
+/**
+ *  \file polynomial_int.h
+ *  Class for Univariate Int Polynomial Base
+ **/
+#ifndef SYMENGINE_UNIVARIATE_INT_H
+#define SYMENGINE_UNIVARIATE_INT_H
+
+#include <symengine/basic.h>
+
+namespace SymEngine
+{
+
+template <typename Coeff, typename Container, typename Poly>
+class UnivariateIntBase : public Basic
+{
+protected:
+    unsigned int degree_;
+    RCP<const Symbol> var_;
+    Container container_;
```

Comment:
Yes, I mixed up `map_uint_mpz` and `map_int_Expr`. The point of a templated function is to write common code. There's nothing common about `map_uint_mpz` and `piranha::polynomial`. How can they be used to write code that is common for both?